modules/tinymce/src/themes/silver/main/ts/ui/skin/Loader.ts

```diff
@@ -23,58 +33,97 @@ const loadRawCss = (editor: Editor, key: string, css: string, styleSheetLoader:
-  if (isInShadowRoot) {
-
-    const skinResourceIdentifier = Options.getSkinUrlOption(editor).getOr('default');
-
-    const shadowDomSkinCss = 'ui/' + skinResourceIdentifier + '/skin.shadowdom.css';
-    const css = tinymce.Resource.get(shadowDomSkinCss);
+  if (!isInShadowRoot) {
+    return Promise.resolve();
+  } else {
+    return loadCSS(
+      editor,
+      'skin.shadowdom',
+      (key, css) => {
+        loadRawCss(editor, key, css, DOMUtils.DOM.styleSheetLoader);
+        return Promise.resolve();
+      },
+      (url) => {
+        return loadStylesheet(editor, url, DOMUtils.DOM.styleSheetLoader);
+      },
+      skinUrl
+    );
 
-    if (Type.isString(css)) {
-      loadRawCss(editor, shadowDomSkinCss, css, DOMUtils.DOM.styleSheetLoader);
-    } else {
-      const suffix = editor.editorManager.suffix;
-      const shadowDomSkinCss = skinUrl + `/skin.shadowdom${suffix}.css`;
-      return loadStylesheet(editor, shadowDomSkinCss, DOMUtils.DOM.styleSheetLoader);
-    }
   }
 };
 
-const loadUrlSkin = async (isInline: boolean, editor: Editor): Promise<void> => {
-  const unbundled = () => {
-    const skinResourceIdentifier = Options.getSkinUrl(editor);
-    const suffix = editor.editorManager.suffix;
-    if (skinResourceIdentifier) {
-      editor.contentCSS.push(skinResourceIdentifier + (isInline ? '/content.inline' : '/content') + `${suffix}.css`);
-    }
-  };
-  Options.getSkinUrlOption(editor).fold(unbundled, (skinUrl) => {
-    const skinContentCss = 'ui/' + skinUrl + (isInline ? '/content.inline' : '/content') + '.css';
-    const css = tinymce.Resource.get(skinContentCss);
-    if (Type.isString(css)) {
-      loadRawCss(editor, skinContentCss, css, editor.ui.styleSheetLoader);
-    } else {
-      unbundled();
-    }
-  });
+const loadUiContentCSS = (editor: Editor, isInline: boolean, skinUrl?: string): Promise<void> =>
+  loadCSS(
+    editor,
+    isInline ? 'content.inline' : 'content',
+    (key, css) => {
+      if (isInline) {
+        loadRawCss(editor, key, css, editor.ui.styleSheetLoader);
+      } else {
+        // Need to wait until the iframe is in the DOM before trying to load
+        // the style into the iframe document
+        editor.on('PostRender', () => {
+          loadRawCss(editor, key, css, editor.dom.styleSheetLoader);
+        });
+      }
+      return Promise.resolve();
+    },
+    (url) => {
+      if (skinUrl) {
+        editor.contentCSS.push(url);
+      }
+      return Promise.resolve();
+    },
+    skinUrl
+  );
 
+const loadUrlSkin = async (isInline: boolean, editor: Editor): Promise<void> => {
   const skinUrl = Options.getSkinUrl(editor);
 
+  await loadUiContentCSS(editor, isInline, skinUrl);
+
   // In Modern Inline, this is explicitly called in editor.on('focus', ...) as well as in render().
   // Seems to work without, but adding a note in case things break later
   if (!Options.isSkinDisabled(editor) && Type.isString(skinUrl)) {
```

Comment:
This suggestion is invalid because the function is declared `async`. That means any non-promise return value will be wrapped in a promise; since all functions have an implicit void return at the end this guarantees a `Promise<void>` return value.